src/Akka.Persistence.Azure/Snapshot/AzureBlobSnapshotStore.cs

```diff
@@ -51,50 +54,49 @@ public AzureBlobSnapshotStore(Config config = null)
 
-            _container = new Lazy<CloudBlobContainer>(() => InitCloudStorage(5).Result);
+            _containerClient = new Lazy<BlobContainerClient>(() => InitCloudStorage(5).Result);
         }
 
-        public CloudBlobContainer Container => _container.Value;
+        public BlobContainerClient Container => _containerClient.Value;
 
-        private async Task<CloudBlobContainer> InitCloudStorage(int remainingTries)
+        private async Task<BlobContainerClient> InitCloudStorage(int remainingTries)
         {
             try
             {
-                var blobClient = _storageAccount.CreateCloudBlobClient();
-                var containerRef = blobClient.GetContainerReference(_settings.ContainerName);
-                var op = new OperationContext();
+                var blobClient = _serviceClient.GetBlobContainerClient(_settings.ContainerName);
 
-                using (var cts = new CancellationTokenSource(_settings.ConnectTimeout))
+                using var cts = new CancellationTokenSource(_settings.ConnectTimeout);
+                if (!_settings.AutoInitialize)
                 {
-                    if (!_settings.AutoInitialize)
-                    {
-                        var exists = await containerRef.ExistsAsync(null, null, cts.Token);
+                    var exists = await blobClient.ExistsAsync(cts.Token);
 
-                        if (!exists)
-                        {
-                            remainingTries = 0;
+                    if (!exists)
+                    {
+                        remainingTries = 0;
 
-                            throw new Exception(
-                                $"Container {_settings.ContainerName} doesn't exist. Either create it or turn auto-initialize on");
-                        }
+                        throw new Exception(
+                            $"Container {_settings.ContainerName} doesn't exist. Either create it or turn auto-initialize on");
+                    }
                         
-                        _log.Info("Successfully connected to existing container", _settings.ContainerName);
+                    _log.Info("Successfully connected to existing container", _settings.ContainerName);
                         
-                        return containerRef;
-                    }
-                    
-                    if (await containerRef.CreateIfNotExistsAsync(BlobContainerPublicAccessType.Container,
-                        new BlobRequestOptions(), op, cts.Token))
-                        _log.Info("Created Azure Blob Container", _settings.ContainerName);
-                    else
-                        _log.Info("Successfully connected to existing container", _settings.ContainerName);
+                    return blobClient;
                 }
 
-                return containerRef;
+                var response = await blobClient.CreateIfNotExistsAsync(
+                    PublicAccessType.BlobContainer,
```

Comment:
Does this affect whether the blob is accessible to the public or is this just weird naming on the part of the Azure SDK?

Reply:
From the documentation:

> Azure Storage supports optional anonymous public read access for containers and blobs. By default, anonymous access to your data is never permitted. Unless you explicitly enable anonymous access, all requests to a container and its blobs must be authorized. When you configure a container's public access level setting to permit anonymous access, clients can read data in that container without authorizing the request.

So the code allows access to the blobs and containers to anonymous connections.
Its there to replace the original code `await containerRef.CreateIfNotExistsAsync(BlobContainerPublicAccessType.Container,
new BlobRequestOptions(), op, cts.Token)` which does the exact same thing.

Reply:
Didn't we just merge a PR in a few days ago that explicitly disabled this (because disallowing anonymous read access to application data by default makes sense) ?

Reply:
I don't see that merge, can you link the PR?

Reply:
It was this issue here: https://github.com/petabridge/Akka.Persistence.Azure/issues/144

Reply:
Looks like they never changed the default setting when `auto-initialize = on`, but it shouldn't be public by default

Reply:
Lets address that in another PR